pkg/ipamd/ipamd.go

```diff
@@ -338,18 +338,41 @@ func (c *IPAMContext) nodeInit() error {
 			time.Sleep(eniAttachTime)
 		}
 	}
-	localPods, err := c.getLocalPodsWithRetry()
+
+	err = c.configureIPRulesForPods(pbVPCcidrs)
 	if err != nil {
-		log.Warnf("During ipamd init, failed to get Pod information from Kubernetes API Server %v", err)
 		ipamdErrInc("nodeInitK8SGetLocalPodIPsFailed")
-		// This can happens when L-IPAMD starts before kubelet.
+		return  err
+	}
+	// For a new node, attach IPs
+	increasedPool, err := c.tryAssignIPs()
+	if err == nil && increasedPool {
+		c.updateLastNodeIPPoolAction()
+	} else {
+		return err
+	}
+
+	//Spawning checkAndUpdateRules go-routine
+	go wait.Forever(func() {
+		log.Debug("Spawning checkAndUpdateRules go-routine")
+		c.checkVPCCIDRsAndRules(pbVPCcidrs)
```

Comment:
How does `pbVPCcidrs` get updated here? I assume we would have to return it from the function, something like:
```
pbVPCcidrs = c.checkVPCCIDRsAndRules(pbVPCcidrs)
```
If not, we will always compare with the initial VPC CIDRs from the first `nodeInit()` call.